src/main/java/io/cdap/plugin/gcp/common/GCPErrorDetailsProvider.java

```diff
@@ -52,6 +53,12 @@ public ProgramFailureException getExceptionDetails(Exception e, ErrorContext err
       if (t instanceof HttpResponseException) {
         return getProgramFailureException((HttpResponseException) t, errorContext);
       }
+      if (t instanceof IllegalArgumentException) {
+        return getProgramFailureException((IllegalArgumentException) t, errorContext);
+      }
+      if (t instanceof IllegalStateException) {
```

Comment:
Is there any place it can in the code still throw `IllegalStateException` after we changed it to `IllegalArgumentException`?

Reply:
There are a few in `BigQueryRecordToJson`

https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java#L105-L106

https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java#L164-L165

https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java#L192-L193

https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java#L205-L206

https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java#L210-L211